src/main/java/org/opensearch/knn/index/mapper/KNNVectorFieldType.java

```diff
@@ -74,11 +91,134 @@ public Query termQuery(Object value, QueryShardContext context) {
+     */
+    public boolean isIndexedForAnn() {
+        return getModelId().isPresent() || getKNNIndexContext().isPresent();
+    }
+
+    /**
+     * Return a map of query parameters that are valid for the given query context and augmented with other
+     * parameters
+     *
+     * @param queryContext Context of the query
+     * @param originalMethodParameters user provided query parameters
+     * @return parameters to be passed to the library augmented based on the field type
+     */
+    public Map<String, Object> getProcessedQueryMethodParameters(QueryContext queryContext, Map<String, Object> originalMethodParameters) {
+        if (originalMethodParameters == null || originalMethodParameters.isEmpty()) {
+            return originalMethodParameters;
+        }
+
+        // If we are unable to get the configuration and the user is trying to passs in parameters, we have to fail
+        // the request
+        KNNIndexContext knnIndexContext = getKNNIndexContext().orElseThrow(
+            () -> new IllegalArgumentException(
+                "Unable to validate passed in method parameters because index was built with model before 2.14"
+            )
+        );
+
+        final KNNLibrarySearchContext engineSpecificMethodContext = knnIndexContext.getKnnLibrarySearchContext();
+        return engineSpecificMethodContext.processMethodParameters(queryContext, originalMethodParameters);
+    }
+
+    public RescoreContext getProcessedRescoreQueryContext(QueryContext queryContext, RescoreContext originalRescoreContext) {
+        if (originalRescoreContext != null) {
+            return originalRescoreContext;
+        }
+        Optional<KNNIndexContext> knnIndexContext = getKNNIndexContext();
+        return knnIndexContext.map(indexContext -> indexContext.getKnnLibrarySearchContext().getDefaultRescoreContext(queryContext))
+            .orElse(RescoreContext.DISABLED_RESCORE_CONTEXT);
+    }
+
+    Optional<KNNIndexContext> getKNNIndexContext() {
+        KNNVectorFieldTypeConfig knnVectorFieldTypeConfig = cachedKNNVectorFieldTypeConfig.getKnnVectorFieldTypeConfig();
+        if (knnVectorFieldTypeConfig == null) {
+            return Optional.empty();
+        }
+        return Optional.ofNullable(knnVectorFieldTypeConfig.getKnnIndexContext());
+    }
+
+    public SpaceType getSpaceType() {
+        return cachedKNNVectorFieldTypeConfig.getKnnVectorFieldTypeConfig().getSpaceType();
+    }
+
+    /**
+     * Configuration class for {@link KNNVectorFieldType}
+     */
+    @AllArgsConstructor
+    @Builder
+    @Getter
+    public static final class KNNVectorFieldTypeConfig {
+        private final int dimension;
+        private final VectorDataType vectorDataType;
+        private final KNNIndexContext knnIndexContext;
+        private final SpaceType spaceType;
+        private final KNNEngine knnEngine;
+    }
+
+    @RequiredArgsConstructor
+    private static class CachedKNNVectorFieldTypeConfig {
```

Comment:
this class seems reduntant , why can't we have suppiler as part of KNNVectorFieldTypeConfig class it self , and that can as wholesome class for caching and noncaching cases.